lib/spack/spack/test/cmd/gc.py

```diff
@@ -39,9 +51,88 @@ def test_gc_with_environment(config, mutable_database, mutable_mock_env_path, ca
     s.package.do_install(fake=True, explicit=True)
 
     e = ev.create("test_gc")
-    e.add("cmake")
     with e:
+        add("cmake")
+        install()
         with capsys.disabled():
+            assert "cmake" in find()
             output = gc("-y")
-    assert "Restricting the garbage collection" in output
+    assert "Restricting garbage collection" in output
     assert "There are no unused specs" in output
+
+
+@pytest.mark.db
+def test_gc_with_build_dependency_in_environment(
+    config, mutable_database, mutable_mock_env_path, capsys
+):
+    s = spack.spec.Spec("simple-inheritance")
+    s.concretize()
+    s.package.do_install(fake=True, explicit=True)
+
+    e = ev.create("test_gc")
+    with e:
+        add("simple-inheritance")
+        install()
+        with capsys.disabled():
+            assert "simple-inheritance" in find()
+            output = gc("-yb")
+    assert "Restricting garbage collection" in output
+    assert "There are no unused specs" in output
+
+    with e:
+        with capsys.disabled():
+            assert "simple-inheritance" in find()
+            output = gc("-y")
+    assert "Restricting garbage collection" in output
+    assert "Successfully uninstalled cmake" in output
+
+
+@pytest.mark.db
+def test_gc_except_any_environments(config, mutable_database, mutable_mock_env_path, capsys):
+    s = spack.spec.Spec("simple-inheritance")
+    s.concretize()
+    s.package.do_install(fake=True, explicit=True)
+
+    assert "zmpi" in find()
```

Comment:
nitpick: I would just use `mutable_database` here

Reply:
addressed in #41876